modules/codegen/src/main/scala/com/twilio/guardrail/ProtocolGenerator.scala

```diff
@@ -169,14 +171,20 @@ object ProtocolGenerator {
     */
   def couldBeSnakeCase(s: String): Boolean = s.toLowerCase(Locale.US) == s
 
-  def getPropertyRequirement(schema: Tracker[Schema[_]], isRequired: Boolean): PropertyRequirement =
+  private[this] def getPropertyRequirement(
+      schema: Tracker[Schema[_]],
+      isRequired: Boolean,
+      defaultPropertyRequirement: PropertyRequirement
+  ): PropertyRequirement =
     (for {
       isNullable <- schema.downField("nullable", _.getNullable)
-    } yield (isRequired, isNullable.getOrElse(false)) match {
-      case (true, false)  => PropertyRequirement.Required
-      case (true, true)   => PropertyRequirement.RequiredNullable
-      case (false, false) => PropertyRequirement.Optional
-      case (false, true)  => PropertyRequirement.OptionalNullable
+    } yield (isRequired, isNullable) match {
+      case (true, None)         => PropertyRequirement.Required
+      case (false, None)        => defaultPropertyRequirement
```

Comment:
im nitpicking here but it might be good to order this somehow to make reading easier ... id suggest to sort by isRequired

Reply:
reordered